Tests/scripts/hook_validations/secrets.py

```diff
@@ -120,39 +126,41 @@ def search_potential_secrets(secrets_file_paths):
     :return: dictionary(filename: (list)secrets) of strings sorted by file name for secrets found in files
     """
     secrets_found = {}
-
-    # Get generic white list set
-    conf_secrets_white_list, ioc_white_list, files_white_list = get_white_list()
-
     for file_path in secrets_file_paths:
+        # Get if file path in pack and pack name
+        is_pack = is_file_path_in_pack(file_path)
+        pack_name = get_pack_name(file_path)
+        # Get generic/ioc/files white list sets based on if pack or not
+        secrets_white_list, ioc_white_list, files_white_list = get_white_listed_items(is_pack, pack_name)
+        # Skip white listed files
         if file_path in files_white_list:
             print("Skipping secrets detection for file: {} as it is white listed".format(file_path))
             continue
-
+        # Init vars for current loop
         file_name = os.path.basename(file_path)
         high_entropy_strings = []
         secrets_found_with_regex = []
         yml_file_contents = None
         _, file_extension = os.path.splitext(file_path)
         skip_secrets = False
-
-        secrets_white_list = set(conf_secrets_white_list)
         # get file contents
         file_contents = get_file_contents(file_path, file_extension)
+        # in packs regard all items as regex as well, reset pack's whitelist in order to avoid repetition later
+        if is_pack:
+            file_contents = remove_white_list_regexs(file_contents, secrets_white_list)
+            secrets_white_list = set()
         # Validate if it is integration documentation file
-        integration_readme = re.match(pattern=INTEGRATION_README_REGEX,
-                                      string=file_path,
-                                      flags=re.IGNORECASE)
-        # if py/js file, search for yml in order to retrieve temp white list
-        if file_extension in {'.py', '.js'} or integration_readme:
+        integration_readme = re.match(pattern=INTEGRATION_README_REGEX, string=file_path, flags=re.IGNORECASE)
```

Comment:
what about `PACK_README_REGEX`?
use `check_type` function

Reply:
I'm a bit conflicted about this, the checked_type we use right now is not for file types more like Demisto types so it's not a good solution for me.
Doing my own check_type would require me to change my file extension set to regexes for each type, is that what we are looking for? Don't see any other use for this data atm. (or to send both file_path & file extension into check type)

Reply:
Went for hybrid solution